src/compiler/scala/tools/nsc/typechecker/ConstantFolder.scala

```diff
@@ -102,13 +102,13 @@ abstract class ConstantFolder {
     case nme.XOR => Constant(x.longValue ^ y.longValue)
     case nme.AND => Constant(x.longValue & y.longValue)
     case nme.LSL if x.tag <= IntTag
-                 => Constant(x.intValue << y.longValue)
+                 => Constant(x.intValue << y.longValue.toInt) // .toInt required to bootstrap despite SI-9516
```

Comment:
maybe change this comment, something like `TODO: remove .toInt once starr is includes the fix for SI-9516 (2.12.0-M5)`